File: Classes/motion/MotionManager.cpp
#include "MotionManager.h"

namespace ju6sigan
{
	MotionManager *MotionManager::m_pInstance;
	
	MotionManager* MotionManager::getInstance()
	{
		if (m_pInstance == NULL)
			m_pInstance = new MotionManager();
		return m_pInstance;
	}
	
	MotionManager::MotionManager()
	{
		m_pSensorHelper = SensorHelper::getInstance();
	}
	
	void MotionManager::enableSensor()
	{
		m_pSensorHelper->enable();
	}
	
	void MotionManager::disableSensor()
	{
		m_pSensorHelper->disable();
	}
	
	void MotionManager::startCapture()
	{
		m_pSensorHelper->start();
	}
	
	Motion MotionManager::stopCapture()
	{
		// Stop capturing sensor data, and get captured data.
		vector<kmVecPair> data;
		m_pSensorHelper->stop(data);
		
		// Recognize and return
		return m_motionRecognizer.recognize(data);
	}
	
	void MotionManager::loadData()
	{
		m_motionRecognizer.load();
	}
}

File: proj.android/src/kr/ac/yonsei/cs/ju6sigan/MotionManager.java
package kr.ac.yonsei.cs.ju6sigan;

public class MotionManager {
	
	public static final int MOTION_HORIZONTAL = 0;
	public static final int MOTION_DIAGONAL = 1;
	public static final int MOTION_CIRCLE = 2;
	public static final int MOTION_UNKNOWN = 4;

	/**
	 * Enable device's sensor.
	 */
	public native void enableSensor();
	
	/**
	 * Disable device's sensor.
	 */
	public native void disableSensor();
	
	/**
	 * Start capturing sensor data.
	 */
	public native void startCapture();
	
	/**
	 * Stop capturing sensor data and recognize.
	 * Types of motion are defined constants.
	 * @return A type of motion
	 */
	public native int stopCapture();
	
}
